taxonomy-editor-frontend/src/pages/editentry/ListAllProperties.jsx

```diff
@@ -1,19 +1,30 @@
 import { IconButton, TextField, Typography } from "@mui/material";
 import AddBoxIcon from '@mui/icons-material/AddBox';
-import { DataGrid } from '@mui/x-data-grid';
+import MaterialTable from '@material-table/core';
+import { useState } from "react";
 
 const ListAllProperties = ({ nodeObject, setNodeObject }) => {
     let toBeRendered = []
+
+    function guidGenerator() {
+        var S4 = function() {
+           return (((1+Math.random())*0x10000)|0).toString(16).substring(1);
+        };
+        return (S4()+S4()+"-"+S4()+"-"+S4()+"-"+S4()+"-"+S4()+S4()+S4());
+    }
+
     Object.keys(nodeObject).forEach((key, index) => {
         if (key.startsWith('prop')) {
                 let property_name = key.split('_').slice(1).join('_');
                 toBeRendered.push({
-                    'id': index+1,
+                    'id': guidGenerator(),
                     'property-name': property_name,
                     'property-value': nodeObject[key]
                 })
             }
     });
+    
+    const [data, setData] = useState(toBeRendered);
```

Comment:
better not call it this way, please prefer to have the hooks before anything else is being called, as they can't be after conditionals, etc. You could instead declare it in line `7` and set it later, possibly in a `useEffect`